src/features/authentication/authSlice.js

```diff
@@ -0,0 +1,268 @@
+let local_token = login?.token || null;
+let local_userData = login?.userData || null;
+let userLoginStatus = local_token ? true : false;
+
+export const authSlice = createSlice({
+  name: "auth",
+  initialState: {
+    status: "idle",
+    error: null,
+    isUserLoggedIn: userLoginStatus,
+    token: local_token,
+    userData: local_userData,
+    shouldUpdatePost: false,
+  },
+  reducers: {
+    setLoginDetails: (state, action) => {
+      state.userData = action.payload.userData;
+      state.token = action.payload.token;
+      state.isUserLoggedIn = true;
+    },
+    logoutUser: (state, action) => {
+      localStorage?.removeItem("ggLogin");
+      state.token = null;
+      state.userData = null;
+      state.isUserLoggedIn = false;
+    },
+    setUserDetails: (state, action) => {
+      state.userData = action.payload.userData;
+      localStorage?.setItem(
+        "ggLogin",
+        JSON.stringify({ userData: state.userData, token: state.token })
+      );
+    },
+    setUpdatePostStatus: (state, action) => {
+      state.shouldUpdatePost = action.payload.updateStatus;
+    },
+  },
+  extraReducers: {
+    [loginUser.pending]: (state) => {
+      state.status = "loading";
+    },
+    [loginUser.fulfilled]: (state, action) => {
+      state.userData = action.payload.user;
+      state.token = action.payload.token;
+      state.isUserLoggedIn = true;
+      state.error = null;
+      state.status = "fulfilled";
+
+      localStorage?.setItem(
+        "ggLogin",
+        JSON.stringify({ userData: state.userData, token: state.token })
+      );
+    },
+    [loginUser.rejected]: (state, action) => {
+      state.status = "error";
+      if (action.payload && action.payload.errorMessage)
+        state.error = action.payload.errorMessage;
+      else state.error = "Something went wrong";
+    },
+    [signupUser.pending]: (state) => {
+      state.status = "loading";
+    },
+    [signupUser.fulfilled]: (state, action) => {
+      state.userData = action.payload.user;
+      state.token = action.payload.token;
+      state.isUserLoggedIn = true;
+      state.error = null;
+      state.status = "fulfilled";
+
+      localStorage?.setItem(
+        "ggLogin",
+        JSON.stringify({ userData: state.userData, token: state.token })
+      );
+    },
+    [signupUser.rejected]: (state, action) => {
+      state.status = "error";
+      if (action.payload && action.payload.errorMessage)
+        state.error = action.payload.errorMessage;
+      else state.error = "Something went wrong";
+    },
+    [updateUserDetails.pending]: (state) => {
+      state.status = "loading";
+    },
+    [updateUserDetails.fulfilled]: (state, action) => {
+      state.userData = action.payload.user;
+      let login = JSON.parse(localStorage.getItem("ggLogin"));
+      if (login) {
+        localStorage?.setItem(
+          "ggLogin",
+          JSON.stringify({ userData: action.payload.user, token: login.token })
+        );
+      }

```

Comment:
can get rid of this operation if you don't store whole user data in localStorage